src/flamegraph/merge.rs

```diff
@@ -83,34 +110,30 @@ where
     let mut last = "";
     let mut tmp = Default::default();
     let mut frames = Default::default();
+    let mut delta = None;
+    let mut delta_max = 1;
     for line in lines {
         let mut line = line.trim();
         if line.is_empty() {
             continue;
         }
 
-        let nsamples = if let Some(samplesi) = line.rfind(' ') {
-            let mut samples = &line[(samplesi + 1)..];
-            // strip fractional part (if any);
-            // foobar 1.klwdjlakdj
-            if let Some(doti) = samples.find('.') {
-                samples = &samples[..doti];
-            }
-            match samples.parse::<usize>() {
-                Ok(nsamples) => {
-                    // remove nsamples part we just parsed from line
-                    line = line[..samplesi].trim_end();
-                    // give out the sample count
-                    nsamples
-                }
-                Err(_) => {
-                    ignored += 1;
-                    continue;
+        // Usually there will only be one samples column at the end of a line,
+        // but for differentials there will be two. In the case there are two
+        // we compute the delta between them and use the second one as as the
+        // number of samples for this frame.
+        let nsamples = match (parse_nsamples(&mut line), parse_nsamples(&mut line)) {
+            (Some(s2), nsamples1) => {
+                delta = nsamples1.map_or(None, |s1| Some(s2 as isize - s1 as isize));
+                if let Some(delta) = delta {
```

Comment:
Given that we have to do this `if let Some` anyway, I think I'd prefer to get rid of the `map` on the line above and instead just do the computation of `delta` inside the `if` too.